### DISCUSSION TITLE
Can `code-server` be used to develop + debug extensions?

### DISCUSSION THREAD

**Discussion by bilogic** [coder/code-server#6685] [Q&A] [answered]
Is it technically possible? If yes, can share references/tutorials etc? Thank you!

**Comment by code-asher** [comment, marked as answer]
I have no references but I opened an extension directory, hit F5, and it opened a new window with the extension and debugger active so it seems to work.

**Reply by bilogic** [reply]
Thanks! Is there a repo for the said extension? TBH, I have never even made an extension before, but would like to give it a shot.

**Reply by code-asher** [reply]
I tested it with our extension here: https://github.com/coder/vscode-coder

But you might want to start here: https://code.visualstudio.com/api/get-started/your-first-extension

**Reply by bilogic** [reply]
Right. I noticed your extension is written in TypeScript, but you were able to hit F5 and debug, where is the transpiler running to compile down to JS?

**Reply by code-asher** [reply]
With our extension I think you have to compile it first (using the terminal for example).  But you can add a pre-launch task to automatically compile on changes.  VS Code's documentation I linked above uses this example: https://github.com/microsoft/vscode-extension-samples/blob/main/helloworld-sample/.vscode/launch.json (note the `npm watch`).

**Reply by bilogic** [reply]
@code-asher 

1. I managed to get it going, but whenever I hit F5, it tries to open the same specific workspace which takes almost a minute. 
2. I tried adding `-n` in `launch.json`'s arg in hope that it will not load the workspace, but that did not help. 
3. Any idea what might be causing my workspace to load?

Thank you!

**Reply by code-asher** [reply]
Oh weird.  When I do it, there is nothing open.  Do you see a `ew=true` query parameter in the URL?  That should force it not to open anything, and when I debug that is automatically added for me.

**Reply by bilogic** [reply]
I don't see `ew=true`, but there it has this `workspace=...` which points to the specific workspace

**Reply by bilogic** [reply]
@code-asher I managed to find the problem. I created a Chrome shortcut to use `code-server` like a desktop VSCode. That shortcut was pointing to the specific workspace. Recreated it without that workspace and it solved my problem. Thank you!

**Reply by code-asher** [reply]
Nice find!  Glad it is working.
